lib/elastic_record/index/manage.rb

```diff
@@ -8,12 +8,12 @@ def create_and_deploy(index_name = new_index_name)
       end
 
       def create(index_name = new_index_name, setting_overrides: {})
-        connection.json_put "/#{index_name}", {
-          "mappings" => {
-            mapping_type => mapping
-          },
+        mapping_params = {
+          "mappings" => (custom_mapping_type_name? ? { mapping_type => mapping } : mapping),
           "settings" => settings.merge(setting_overrides)
         }
+
+        connection.json_put "/#{index_name}?include_type_name=#{custom_mapping_type_name?}", mapping_params
```

Comment:
Perhaps add:
```suggestion
        # TODO: Remove include_type_name when ES8 support is added
        connection.json_put "/#{index_name}?include_type_name=#{custom_mapping_type_name?}", mapping_params
```

Reply:
How about:

`TODO: Evaluate necessity of this param during ES8 update`?

It was necessary to provide it to bypass some kind of error checking ES7 was doing, even thought the format of the mapping seemed to map their new structure + the param is default faulse. I guess I just want to keep it backwards compatible a version to eliminate update woes.

Reply:
My wording was because of what I read in the removal of mapping types link you had:

> **Elasticsearch 8.x**
> * The `include_type_name` parameter is removed.

Reply:
Right, but then it's usage should be conditional vs straight up removed if we want the gem to work with both versions while we update ES.

Reply:
eh, it's a comment not worth arguing about. We can always update it.